operational_analysis/methods/turbine_long_term_gross_energy.py

```diff
@@ -77,30 +81,53 @@ def __init__(self, plant, max_power_filter = 0.85, wind_bin_thresh=2):
         # Set number of 'valid' counts required when summing data to daily values
         self._num_valid_daily = 60. / self._time_conversion[self._plant._scada_freq] * 24
         
+
+        
         # Initially sort the different turbine data into dictionary entries
+        logger.info("Processing SCADA data into dictionaries by turbine (this can take a while)")
         self.sort_scada_by_turbine()
         
 
     @logged_method_call
-    def run(self, reanal_subset):
+    def run(self, reanal_subset = ['erai', 'ncep2', 'merra2'], max_power_filter = 0.85, 
+            wind_bin_thresh = 2, correction_threshold = 0.90, enable_plotting = False,
+            plot_dir = None):
         """
         Perform pre-processing of data into an internal representation for which the analysis can run more quickly.
         
         Args:
-            (None)
+            reanal_subset(:obj:`list`): Which reanalysis products to use for long-term correction
+            max_power_filter(:obj:`float`): Maximum power threshold (fraction) to which the bin filter 
+                                            should be applied (default 0.85)
+            wind_bin_thresh(:obj:`float`): The filter threshold for each bin (default is 2 m/s)
+            correction_threshold(:obj:`float`): The threshold (fraction) above which daily scada energy data
+                                                should be corrected (default is 0.90)
+            enable_plotting(:obj:`boolean`): Indicate whether to output plots
+            plot_dir(:obj:`string`): Location to save figures
             
         Returns:
             (None)
         """
         
-        self._reanal = reanal_subset
+        # Assign parameters as object attributes
+        self._reanal = reanal_subset # Reanalysis data to consider in fitting
+        self._max_power_filter = max_power_filter 
+        self._wind_bin_thresh = wind_bin_thresh
+        self._correction_threshold = correction_threshold
         
         logger.info("Filtering turbine data")
         self.filter_turbine_data() # Filter turbine data
         
+        if enable_plotting:
+            logger.info("Plotting filtered power curves")
+            self.plot_filtered_power_curves(plot_dir) # Setup daily reanalysis products
+        
         logger.info("Processing reanalysis data to daily averages")
         self.setup_daily_reanalysis_data() # Setup daily reanalysis products
         
+        logger.info("Processing scada data to daily sums")
```

Comment:
I split out processing of daily scada from setup_model_dict() into its own function since more steps are now included (e.g. imputing)